### ISSUE TITLE
500 error on /feeds/new page

### ISSUE THREAD

**Issue by snoop168** [concerto/concerto#1174]
Getting a 500 error. Other pages seem to load fine. my apache error.log file shows:

```
App 3622 stderr: /webapps/concerto/lib/command_check.rb:6: warning: Insecure world writable dir /webapps/concerto/vendor/bundle/ruby/1.9.1/bin in PATH, mode 040777
App 3622 stderr: /usr/local/bin/bundle:23:in `load'
App 3622 stderr: : cannot load such file -- /usr/lib/ruby/bin/bundle (LoadError)
App 3622 stderr: 	from /usr/local/bin/bundle:23:in `<main>'
```
Is it a problem with my bundler? the path /usr/lib/ruby/bin/ does not exist on my machine.

![image](https://cloud.githubusercontent.com/assets/2731594/5918139/4bbb64ba-a5f3-11e4-8c0b-1eb0ea5b605e.png)

**Comment by zr2d2** [comment]
I would start by changing /webapps/concerto/vendor/bundle/ruby/1.9.1/bin
with `chmod go-w /webapps/concerto/vendor/bundle/ruby/1.9.1/bin`

Getting a 500 error. Other pages seem to load fine. my apache error.log
file shows:

App 3622 stderr: /webapps/concerto/lib/command_check.rb:6: warning:
Insecure world writable dir
/webapps/concerto/vendor/bundle/ruby/1.9.1/bin in PATH, mode 040777
App 3622 stderr: /usr/local/bin/bundle:23:in `load'
App 3622 stderr: : cannot load such file -- /usr/lib/ruby/bin/bundle (LoadError)
App 3622 stderr:    from /usr/local/bin/bundle:23:in `<main>'

Is it a problem with my bundler? the path /usr/lib/ruby/bin/ does not exist
on my machine.

[image: image]
<https://cloud.githubusercontent.com/assets/2731594/5918139/4bbb64ba-a5f3-11e4-8c0b-1eb0ea5b605e.png>

—
Reply to this email directly or view it on GitHub
<https://github.com/concerto/concerto/issues/1174>.

**Comment by snoop168** [comment]
```
[Tue Jan 27 08:16:42.859079 2015] [mpm_prefork:notice] [pid 3587] AH00169: caught SIGTERM, shutting down
[ 2015-01-27 08:16:43.9351 7460/7f7738146780 agents/Watchdog/Main.cpp:538 ]: Options: { 'analytics_log_user' => 'nobody', 'default_group' => 'nogroup', 'default_python' => 'python', 'default_ruby' => '/usr/bin/ruby1.9.1', 'default_user' => 'nobody', 'log_level' => '0', 'max_pool_size' => '6', 'passenger_root' => '/var/lib/gems/1.9.1/gems/passenger-4.0.46', 'passenger_version' => '4.0.46', 'pool_idle_time' => '300', 'temp_dir' => '/tmp', 'union_station_gateway_address' => 'gateway.unionstationapp.com', 'union_station_gateway_port' => '443', 'user_switching' => 'true', 'web_server_passenger_version' => '4.0.46', 'web_server_pid' => '7459', 'web_server_type' => 'apache', 'web_server_worker_gid' => '33', 'web_server_worker_uid' => '33' }
[ 2015-01-27 08:16:43.9400 7463/7f97fd326780 agents/HelperAgent/Main.cpp:649 ]: PassengerHelperAgent online, listening at unix:/tmp/passenger.1.0.7459/generation-0/request
[ 2015-01-27 08:16:43.9478 7468/7f28d0b43780 agents/LoggingAgent/Main.cpp:321 ]: PassengerLoggingAgent online, listening at unix:/tmp/passenger.1.0.7459/generation-0/logging
[ 2015-01-27 08:16:43.9484 7460/7f7738146780 agents/Watchdog/Main.cpp:728 ]: All Phusion Passenger agents started!
[ 2015-01-27 08:16:43.9916 7484/7fb528171780 agents/Watchdog/Main.cpp:538 ]: Options: { 'analytics_log_user' => 'nobody', 'default_group' => 'nogroup', 'default_python' => 'python', 'default_ruby' => '/usr/bin/ruby1.9.1', 'default_user' => 'nobody', 'log_level' => '0', 'max_pool_size' => '6', 'passenger_root' => '/var/lib/gems/1.9.1/gems/passenger-4.0.46', 'passenger_version' => '4.0.46', 'pool_idle_time' => '300', 'temp_dir' => '/tmp', 'union_station_gateway_address' => 'gateway.unionstationapp.com', 'union_station_gateway_port' => '443', 'user_switching' => 'true', 'web_server_passenger_version' => '4.0.46', 'web_server_pid' => '7480', 'web_server_type' => 'apache', 'web_server_worker_gid' => '33', 'web_server_worker_uid' => '33' }
[ 2015-01-27 08:16:43.9969 7487/7f1376ce8780 agents/HelperAgent/Main.cpp:649 ]: PassengerHelperAgent online, listening at unix:/tmp/passenger.1.0.7480/generation-0/request
[ 2015-01-27 08:16:44.0064 7492/7f79a52b9780 agents/LoggingAgent/Main.cpp:321 ]: PassengerLoggingAgent online, listening at unix:/tmp/passenger.1.0.7480/generation-0/logging
[Tue Jan 27 08:16:44.009659 2015] [mpm_prefork:notice] [pid 7480] AH00163: Apache/2.4.7 (Ubuntu) PHP/5.5.9-1ubuntu4.5 Phusion_Passenger/4.0.46 configured -- resuming normal operations
[Tue Jan 27 08:16:44.009721 2015] [core:notice] [pid 7480] AH00094: Command line: '/usr/sbin/apache2'
[ 2015-01-27 08:16:44.0097 7484/7fb528171780 agents/Watchdog/Main.cpp:728 ]: All Phusion Passenger agents started!
App 7533 stdout: 
App 7533 stderr: [deprecated] I18n.enforce_available_locales will default to true in the future. If you really want to skip validation of your locale you can set I18n.enforce_available_locales = false to avoid this message.
App 7533 stderr: /webapps/concerto/lib/command_check.rb:6: warning: Insecure world writable dir /webapps/concerto/vendor/bundle/ruby/1.9.1 in PATH, mode 040777
[ 2015-01-27 08:17:00.6120 7487/7f1376c02700 Pool2/SmartSpawner.h:298 ]: Preloader for /webapps/concerto started on PID 7533, listening on unix:/tmp/passenger.1.0.7480/generation-0/backends/preloader.7533
App 7557 stdout: 
App 7533 stderr: /webapps/concerto/lib/command_check.rb:6: warning: Insecure world writable dir /webapps/concerto/vendor/bundle/ruby/1.9.1 in PATH, mode 040777
App 7533 stderr: /usr/local/bin/bundle:23:in `load'
App 7533 stderr: : 
App 7533 stderr: cannot load such file -- /usr/lib/ruby/bin/bundle
App 7533 stderr:  (
App 7533 stderr: LoadError
App 7533 stderr: )
App 7533 stderr: 	from /usr/local/bin/bundle:23:in `<main>'
App 7533 stderr: /webapps/concerto/lib/command_check.rb:6: warning: Insecure world writable dir /webapps/concerto/vendor/bundle/ruby/1.9.1 in PATH, mode 040777
App 7533 stderr: /usr/local/bin/bundle:23:in `load'
App 7533 stderr: : cannot load such file -- /usr/lib/ruby/bin/bundle (LoadError)
App 7533 stderr: 	from /usr/local/bin/bundle:23:in `<main>'
```

**Comment by snoop168** [comment]
so ran the chmod and the above error is still received. Why is it trying to load a file in /usr/lib/ruby/bin/bundle?

I looked at line 23 in the /usr/local/bin/bundle file which seems to be whats calling this file:
```
load Gem.bin_path('bundler', 'bundle', version)
```

**Comment by snoop168** [comment]
@zr2d2 Any other ideas? 

from my understanding `Gem.bin_path` returns the path of the gem, I assume `load` runs it. Why would `Gem.bin_path` return a path that doesn't exist?

**Comment by bamnet** [comment]
I find it weird that this only happens on /feeds/new.  That's a pretty vanilla action in the controller and i don't see any particular reason why we'd be calling out to bundler on that page vs another.

* Does this reproduce both in development and production environments?
* Is there anything in the Rails logs for these requests?

**Comment by snoop168** [comment]
so for me, running the dev server on port 3000 yields slightly different results. I have my dev DB set to the same DB as the prod one for now. When I load that page i just get the diagonal lined background with nothing else. My dev.log file shows the following entries for the page load:

```
Started GET "/feeds/new" for 10.35.4.30 at 2015-01-29 08:30:31 -0500
Processing by FeedsController#new as HTML
  [1m[36mUser Load (0.1ms)[0m  [1mSELECT  `users`.* FROM `users`  WHERE `users`.`id` = 1  ORDER BY `users`.`id` ASC LIMIT 1[0m
  [1m[35mGroup Load (0.1ms)[0m  SELECT `groups`.* FROM `groups` INNER JOIN `memberships` ON `groups`.`id` = `memberships`.`group_id` WHERE `memberships`.`level` = 9 AND `memberships`.`user_id` = 1
  [1m[36mFeed Load (0.1ms)[0m  [1mSELECT `feeds`.* FROM `feeds`  WHERE `feeds`.`group_id` = 1[0m
  [1m[35mSubmission Load (0.1ms)[0m  SELECT `submissions`.* FROM `submissions`  WHERE `submissions`.`feed_id` = 1 AND (moderation_flag IS NULL)
  [1m[36mConcertoConfig Load (0.2ms)[0m  [1mSELECT  `concerto_configs`.* FROM `concerto_configs`  WHERE `concerto_configs`.`key` = 'config_last_updated'  ORDER BY `concerto_configs`.`id` ASC LIMIT 1[0m
Cache hit on system_time_zone -  Eastern Time (US & Canada)
  [1m[35mCACHE (0.0ms)[0m  SELECT  `concerto_configs`.* FROM `concerto_configs`  WHERE `concerto_configs`.`key` = 'config_last_updated'  ORDER BY `concerto_configs`.`id` ASC LIMIT 1
Cache hit on public_concerto -  true
  [1m[36mCACHE (0.0ms)[0m  [1mSELECT  `concerto_configs`.* FROM `concerto_configs`  WHERE `concerto_configs`.`key` = 'config_last_updated'  ORDER BY `concerto_configs`.`id` ASC LIMIT 1[0m
Cache hit on allow_registration -  true
  [1m[35mCACHE (0.0ms)[0m  SELECT  `concerto_configs`.* FROM `concerto_configs`  WHERE `concerto_configs`.`key` = 'config_last_updated'  ORDER BY `concerto_configs`.`id` ASC LIMIT 1
Cache hit on allow_user_screen_creation -  false
  [1m[36m (0.1ms)[0m  [1mSELECT `groups`.id FROM `groups` INNER JOIN `memberships` ON `groups`.`id` = `memberships`.`group_id` WHERE (memberships.level > 1) AND `memberships`.`user_id` = 1[0m
  [1m[35mCACHE (0.0ms)[0m  SELECT `groups`.id FROM `groups` INNER JOIN `memberships` ON `groups`.`id` = `memberships`.`group_id` WHERE (memberships.level > 1) AND `memberships`.`user_id` = 1  [["user_id", 1]]
  [1m[36mCACHE (0.0ms)[0m  [1mSELECT `groups`.id FROM `groups` INNER JOIN `memberships` ON `groups`.`id` = `memberships`.`group_id` WHERE (memberships.level > 1) AND `memberships`.`user_id` = 1[0m  [["user_id", 1]]
  [1m[35mCACHE (0.0ms)[0m  SELECT  `concerto_configs`.* FROM `concerto_configs`  WHERE `concerto_configs`.`key` = 'config_last_updated'  ORDER BY `concerto_configs`.`id` ASC LIMIT 1
Cache hit on allow_user_feed_creation -  true
  [1m[36mCACHE (0.0ms)[0m  [1mSELECT `groups`.id FROM `groups` INNER JOIN `memberships` ON `groups`.`id` = `memberships`.`group_id` WHERE (memberships.level > 1) AND `memberships`.`user_id` = 1[0m  [["user_id", 1]]
  [1m[35mCACHE (0.0ms)[0m  SELECT `groups`.id FROM `groups` INNER JOIN `memberships` ON `groups`.`id` = `memberships`.`group_id` WHERE (memberships.level > 1) AND `memberships`.`user_id` = 1  [["user_id", 1]]
  Rendered layouts/_errors.html.erb (0.9ms)
  [1m[36mFeed Load (0.2ms)[0m  [1mSELECT `feeds`.* FROM `feeds`  WHERE `feeds`.`parent_id` IS NULL[0m
  [1m[35mFeed Exists (0.1ms)[0m  SELECT  1 AS one FROM `feeds`  WHERE `feeds`.`parent_id` = 1 LIMIT 1
  Rendered feeds/_form.html.erb (171.6ms)
  Rendered feeds/new.html.erb within layouts/application (173.2ms)
Completed 200 OK in 251ms (ActiveRecord: 1.2ms)
```

When you asked for what my "rails logs" have, is the production and development.log files in the concerto/log folder the rails logs, or are there other logs?

**Comment by snoop168** [comment]
When looking at the logs and comparing to other page loads, the "/feeds/new" request only renders 3 pages:
layouts/_errors.html.erb
feeds/_form.html.erb
feeds/new.html.erb within layouts/application

Comparing to other pages (obviously they probably are not the same by design) but there are significantly more and _errors.html.erb is not one that the others are trying to load.

Does that help? I guess if you can advise where the code would be that would decide what pages to load i can look into this further. I just haven't yet figured out where the code entry point is for a single page load in ruby and how it calls to other files just yet.. (work in progress - basically self learning from this project) If you could point me towards the entry point file I can try and help...

**Comment by xd1936** [comment]
Also having this issue on our test machine that we upgraded to 2.2.5. Let me know if there's anything I can do to help test.

**Comment by romoloman** [comment]
Also here after upgrade to 2.2.5 new feeds display a blank page.
Any way to roll back to a previous version using apt-get ?

**Comment by xd1936** [comment]
What are the ramifications of making this change?

**Comment by augustf** [comment]
The primary issue with #1177 as it is right now is that just doing Group.all allows users to see and exercise some minor control over groups they may have no access to. What needs to happen is that the array of groups needs to be limited to ones the user should be seeing. Normally, accessible_by should do that, but I think that idiosyncrasies in our CanCan implementation make only our custom auth! method usable for that -- and I've not yet found the appropriate documentation for how it works in this situation.

**Comment by romoloman** [comment]
Not particular ramification, is a partial rollback of a previous commit.
So we go back to a non completely correct implementation that anyway makes concerto usable.

**Comment by simplysoft** [comment]
As a workaround you can also disable public concerto (if you are not relying on it), or you can replace https://github.com/concerto/concerto/blob/master/app/models/ability.rb#L52-57 with ```can :read, Group```

we are working on a fix for the root cause (see #1177)
